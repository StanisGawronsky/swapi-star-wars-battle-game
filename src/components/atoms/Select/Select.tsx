import React from "react";

interface IProps {
  onChange: (value: string) => void;
  options: string[];
  label: string;
}

const Select = ({ onChange, options, label }: IProps) => {
  return (
    <div>
      <label>{label}</label>
      <select
        onChange={(e) => {
          onChange(e.target.value as string);
        }}
      >
        {options?.map((type: string) => {
          return (
            <option value={type} key={type}>
              {type}
            </option>
          );
        })}
      </select>
    </div>
  );
};

export default Select;
