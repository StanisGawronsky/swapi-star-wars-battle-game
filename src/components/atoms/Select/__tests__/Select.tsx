import React from "react";
import { render } from "@testing-library/react";
import Select from "../Select";
const options = ["1", "2"];
describe("<Select />", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Select onChange={jest.fn()} options={options} label="1" />
    );
    expect(container).toMatchSnapshot();
  });
});
