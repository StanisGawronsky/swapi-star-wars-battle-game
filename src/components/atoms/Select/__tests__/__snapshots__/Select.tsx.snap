// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly 1`] = `
<div>
  <div>
    <label>
      1
    </label>
    <select>
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
    </select>
  </div>
</div>
`;
